clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -6691,6 +6692,127 @@ test_options_match()
     test.check_options();
 }
 
+// A mock noise generator that only generates TRACE_TYPE_READ records with
+// address 0xdeadbeef.
+class mock_noise_generator_t : public noise_generator_t {
+public:
+    mock_noise_generator_t() {};
+
+    mock_noise_generator_t(uint64_t num_records_to_generate, addr_t addr_to_generate)
+        : num_records_to_generate_(num_records_to_generate)
+        , addr_to_generate_(addr_to_generate) {};
+
+protected:
+    trace_entry_t *
+    read_next_entry() override
+    {
+        if (num_records_to_generate_ == 0) {
+            at_eof_ = true;
+            return nullptr;
+        }
+
+        // Do not change the order for generating TRACE_TYPE_THREAD and TRACE_TYPE_PID.
```

Comment:
Could we put this in its own function to share w/ the parent and avoid duplicating here?

Reply:
Yes, moved the logic on what noise-records to generate to `generate_trace_entry()`.